dissect/target/plugins/os/unix/linux/network.py

```diff
@@ -0,0 +1,324 @@
+
+                dns_value = to_list(network_section.get("DNS", []))
+                dns.update({self._parse_dns_ip(dns_ip) for dns_ip in dns_value})
+
+                address_value = to_list(network_section.get("Address", []))
+                ip_interfaces.update({ip_interface(addr) for addr in address_value})
+
+                gateway_value = to_list(network_section.get("Gateway", []))
+                gateways.update({ip_address(gateway) for gateway in gateway_value})
+
+                vlan_names = network_section.get("VLAN", [])
+                vlan_ids = {
+                    vlan_id
+                    for vlan_name in to_list(vlan_names)
+                    if (vlan_id := virtual_networks.get(vlan_name)) is not None
+                }
+
+                # There are possibly multiple route sections, but they are collapsed into one by the parser.
+                route_section = config.get("Route", {})
+                gateway_values = to_list(route_section.get("Gateway", []))
+                gateways.update(filter(None, map(self._parse_gateway, gateway_values)))
+
+                dhcp_ipv4, dhcp_ipv6 = self._parse_dhcp(network_section.get("DHCP"))
+
+                yield UnixInterfaceRecord(
+                    source=str(config_file),
+                    type=match_section.get("Type", None),
+                    enabled=None,  # Unknown, dependent on run-time state
+                    dhcp_ipv4=dhcp_ipv4,
+                    dhcp_ipv6=dhcp_ipv6,
+                    name=match_section.get("Name", None),
+                    dns=list(dns),
+                    mac=list(mac_addresses),
+                    ip=[interface.ip for interface in ip_interfaces],
+                    network=[interface.network for interface in ip_interfaces],
+                    gateway=list(gateways),
+                    vlan=list(vlan_ids),
+                    configurator="systemd-networkd",
+                )
+            except Exception as e:
+                self._target.log.warning("Error parsing network config file %s", config_file, exc_info=e)
+
+    def _parse_dns_ip(self, address: str) -> ip_address:
+        """Parse DNS address from systemd network configuration file.
+
+        The optional brackets and port number make this hard to parse.
+        See https://www.freedesktop.org/software/systemd/man/latest/systemd.network.html and search for DNS.
+        """
+
+        match = self.dns_ip_patttern.search(address)
+        if not match:
+            raise ValueError(f"Invalid DNS address format: {address}")
+
+        return ip_address(match.group("withoutBrackets") or match.group("withBrackets"))
+
+    def _parse_dhcp(self, value: str | None) -> DhcpConfig:
+        """Parse DHCP value from systemd network configuration file to a named tuple (ipv4, ipv6)."""
+
+        if value is None or value == "no":
+            return self.DhcpConfig(ipv4=False, ipv6=False)
```

Comment:
The named tuple is scoped to the class, reflecting its specificity.

Arguably making the code explicit is favorable to writing a comment explaining it.
Although these are not mutually exclusive,  since this is a private method a single line docstring should be adequate.